### PR TITLE
Make endpoint to upload hood

### PR DESCRIPTION
**Pull request by hectron**

### CODE REVIEW

File: app/controllers/onboarding_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    entries = permitted_params.to_h[:entries]
+
+    Hood::Onboarder.run(entries)
+  end
+
+  private
+
+  def authenticate
+    authenticate_or_request_with_http_token do |token, options|
```
Comment (hectron) [inline app/controllers/onboarding_controller.rb:15]: This ensures that there is a Bearer token set.

For more info, https://guides.rubyonrails.org/action_controller_overview.html#http-token-authentication

File: app/services/hood/onboarder.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Hood
   class Onboarder
-    RequiredHeaders = %i(
+    RequiredFields = %i(
```
Comment (hectron) [inline app/services/hood/onboarder.rb:3]: Converted this to be more generic and support an array of hashes (which could come from parsing a CSV into that, or a JSON object into that).

File: app/controllers/onboarding_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class OnboardingController < ApplicationController
+  before_action :authenticate
+
+  def hood_and_houses
+    permitted_params = params.permit(entries: Hood::Onboarder::RequiredFields)
```
Comment (hectron) [inline app/controllers/onboarding_controller.rb:5]: Example request that:

- Creates two neighborhoods ("Schitt's Creek" and "Star City").
- Creates two houses in "Star City".

```json
{
  "entries": [
    {
      "hood_name": "Schitt's Creek",
      "hood_zip_code": "60601",
      "house_street_address": "",
      "house_city": "",
      "house_state": "",
      "house_zip_code": "",
      "house_bedrooms": "",
      "house_bathrooms": "",
      "house_square_feet": ""
    },
    {
      "hood_name": "Star City",
      "hood_zip_code": "60622",
      "house_street_address": "123 Fake St",
      "house_city": "Chicago",
      "house_state": "IL",
      "house_zip_code": "60622",
      "house_bedrooms": "3",
      "house_bathrooms": "1",
      "house_square_feet": "1200"
    },
    {
      "hood_name": "Star City",
      "hood_zip_code": "60622",
      "house_street_address": "125 Fake St",
      "house_city": "Chicago",
      "house_state": "IL",
      "house_zip_code": "60622",
      "house_bedrooms": "2",
      "house_bathrooms": "2",
      "house_square_feet": "1000"
    }
  ]
}
```

File: app/controllers/onboarding_controller.rb (hunk 1 above)
Comment (hebron-george) [reply app/controllers/onboarding_controller.rb:15]: So the client will have to send this token as part of the request header, got it 👍🏽

File: app/controllers/onboarding_controller.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    entries = permitted_params.to_h[:entries]
```
Comment (hebron-george) [inline app/controllers/onboarding_controller.rb:7]: Isn't `permitted_params` already a hash-like object? Not a blocker

**Review by hebron-george — APPROVED** [review]

**Pull request merged**